envoy/pkg/livenessprobe/cmd/root.go

```diff
@@ -0,0 +1,188 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/cybozu-go/log"
+	"github.com/cybozu-go/well"
+	"github.com/spf13/cobra"
+)
+
+var config struct {
+	timeout    time.Duration
+	listenAddr string
+	readyURL   string
+	httpURL    string
+	httpsAddr  string
+}
+
+type monitor struct {
+	client    *http.Client
+	timeout   time.Duration
+	readyURL  string
+	httpURL   string
+	httpsAddr string
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "livenessprobe",
+	Short: "liveness probe for Envoy",
+	Long:  `Liveness probe for Envoy.`,
+
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		cmd.SilenceUsage = true
```

Comment:
This is only needed when you use `RunE` instead of `Run`.

Reply:
Also, using `PersistentPreRun` is a little overkill because there is only one
subcommand (rootCmd).  The logic here can be moved inside `Run`.